### PR TITLE
Add Tether component documentation

### PR DESCRIPTION
**Pull request by farazcsk**

### CODE REVIEW

File: styleguide/screens/Patterns/Tether/Tether.css [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+}
+
+.controls div:not(:last-of-type) {
+  margin-bottom: 1rem;
+}
+
+.controls div label {
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Tether/Tether.css]: Why can't we add a class to the label directly?

File: styleguide/screens/Patterns/Tether/Tether.css [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+.controls {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+}
+
+.controls div:not(:last-of-type) {
+  margin-bottom: 1rem;
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Tether/Tether.css]: We usually try to use `margin-top` to do vertical padding. Could this work?

```css
.control + .control {
  margin-top: var(--size-regular);
}
```

(It should also use the variables)

File: styleguide/screens/Patterns/Tether/Tether.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+              target={ <Target /> }
+              verticalAttachment={ verticalAttachment }
+              horizontalAttachment={ horizontalAttachment }
+              flushVertical={ flushVertical }
+              flushHorizontal={ flushHorizontal }
+              active
+            >
+              <TetherChild className={ css.tether } />
+            </Tether>
+            <div className={ css.controls } >
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Tether/Tether.js]: I don't think the controls should be inside the specimen's container. Does it not make more sense to have them horizontally across the top of the container? I also think we can use our own components in this case else it'll look a bit naff

File: styleguide/screens/Patterns/Tether/Tether.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+                  onChange={ this.handleFlushHorizontalChange }
+                />
+              </div>
+            </div>
+          </Specimen>
+          <T elm="p" className={ m.mtr }>
+            To position the <C>Tether</C> element provide the <C>verticalAttachment</C> and
+            <C>horizontalAttachment</C> props to the <C>Tether</C> component.
+          </T>
+          <T elm="p" className={ m.mtr }>
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Tether/Tether.js]: This bit is really hard to read with so many code snippets. Thinking that perhaps we should reevaluate how we list out the possible values for each attachment

Comment (farazcsk) [reply styleguide/screens/Patterns/Tether/Tether.js]: Agreed. Do you think we still need to list them out now that we have an interactive example?

Comment (rdjpalmer) [reply styleguide/screens/Patterns/Tether/Tether.js]: Probably not 👍

File: styleguide/screens/Patterns/Tether/Tether.js [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+import Select from '../../../../components/Form/Select/Select';
+import Option from '../../../../components/Form/Select/Option';
+import Checkbox from '../../../../components/Form/Checkbox/Checkbox';
+import { Label } from '../../../../components/Form/FormComponents';
+
+import m from '../../../../globals/modifiers.css';
+import css from './Tether.css';
+import scaffoldCss from '../../../components/Scaffold/Scaffold.css';
+
+const Target = () => <button>Anchor</button>;
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Tether/Tether.js]: Any particular reason you decided to use a button for the target?

Comment (farazcsk) [reply styleguide/screens/Patterns/Tether/Tether.js]: Just based on the story we have already have for the `Tether` component, happy to change it.

File: styleguide/screens/Patterns/Tether/Tether.css [hunk 6]
```diff
@@ -1,29 +1,40 @@
...
+
 .tether {
-  height: 3.75rem;
-  width: 3.75rem;
+  height: var(--size-lg-ii);
+  width: var(--size-lg-ii);
   background-color: var(--color-gold);
 }
+
+@media(min-width: 35rem) {
```
Comment (rdjpalmer) [inline styleguide/screens/Patterns/Tether/Tether.css]: variables

**Pull request merged**